src/uagents/protocol.py

```diff
@@ -176,7 +176,7 @@ def manifest(self) -> Dict[str, Any]:
 
         for schema_digest, model in all_models.items():
             manifest["models"].append(
-                {"digest": schema_digest, "schema": model.schema()}
+                {"digest": schema_digest, "schema": model.schema_no_descriptions}
```

Comment:
Is it guaranteed that this value won't be None?

Reply:
This class attribute is set when generating model digest and we calculate the digest for every model beforehand but for safety I can modify this line to:
`model.schema_no_descriptions if model.schema_no_descriptions is not None else model.schema()`
Does it make sense?

Reply:
In the refactored version I call the schema_no_descriptions(), that method also uses a class attribute as a cache and that class attribute could possibly be None, but in the schema_no_descriptions() method I guarantee to set the value of that class variable now if that is None.